packages/web/graphql/getPatrons.ts

```diff
@@ -0,0 +1,78 @@
+import { BigNumber } from 'ethers';
+import gql from 'fake-tag';
+
+import {
+  GetPatronsQuery,
+  GetPatronsQueryVariables,
+  PlayerFragmentFragment,
+} from './autogen/types';
+import { client } from './client';
+import { PlayerFragment } from './fragments';
+
+const patronsQuery = gql`
+  query GetPatrons($offset: Int) {
+    player(
+      where: { ethereum_address: { _neq: "" }, total_xp: { _gt: 0 } }
+      limit: 50
+      offset: $offset
+    ) {
+      ...PlayerFragment
+    }
+  }
+  ${PlayerFragment}
+`;
+
+export const getPatronsWithOffset = async (
+  offset = 0,
+): Promise<Array<PlayerFragmentFragment>> => {
+  const { data, error } = await client
+    .query<GetPatronsQuery, GetPatronsQueryVariables>(patronsQuery, {
+      offset,
+    })
+    .toPromise();
+
+  if (!data) {
+    if (error) {
+      throw error;
+    }
+
+    return [];
+  }
+
+  return data.player;
+};
+
+const pSeedDescSort = (
+  playerA: PlayerFragmentFragment,
+  playerB: PlayerFragmentFragment,
+) => {
+  const balanceA = BigNumber.from(playerA.token_balance?.pSeedBalance || 0);
+  const balanceB = BigNumber.from(playerB.token_balance?.pSeedBalance || 0);
+
+  if (balanceA.lt(balanceB)) {
+    return 1;
+  } if (balanceA.gt(balanceB)) {
+    return -1;
+  } 
+    return 0;
+  
+};
+
+export const getPatrons = async () => {
+  let patrons: Array<PlayerFragmentFragment> = [];
+  let page = 0;
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+  // eslint-disable-next-line no-await-in-loop
+    const offsettedPatrons = await getPatronsWithOffset(page * 50);
+    patrons = offsettedPatrons.concat(patrons);
+    if (offsettedPatrons.length < 50) break;
+    page += 1;
+  }
+
+  return patrons
+    .filter((p) => BigNumber.from(p.token_balance?.pSeedBalance || 0).gt(0))
+    .sort(pSeedDescSort)
```

Comment:
I could not do all of this on the backend. Do let me know if there is a way we can achieve this in hasura itself. 

Having said that these calls will be made only during static site generation so I think it should be fine.

Reply:
As the token balance is a remote schema, hasura has to query TheGraph for every user and get its balances. One of this query will be super long to process (here 50 calls to TheGraph each loop)

Yes, it will be done only once at static generation, but maybe we should find another way of doing that.

Maybe we can query directly the Seed Graph to get a sorted list of patrons addresses, then get the corresponding players ?

Reply:
Valid points. Firstly I was just checking, the `ethereum_address` stored in the backend is not in lowercase, and theGraph always gives us lowercase addresses. And we can't query our database using that. 

So first I'm creating a separate issue to address that before we can go ahead with this. 

https://github.com/MetaFam/TheGame/issues/404